### PR TITLE
Remove compatibility checks for major version

### LINKED ISSUE #28806: Keep minor and major version announcements consistent
On the [WooCommerce 4.9.0 release post](https://developer.woocommerce.com/2021/01/12/woocommerce-4-9-0-released/) we can read:

> This is a **minor release**, which means everything is backward compatible with the previous version.

But when updating from 4.8.0 to 4.9.0 we get this warning on WP Admin:

image

I understand this is triggered probably by the fact the second part of the version number changes, and it's actually not a big deal, but It might get confusing for some users.

### PR DESCRIPTION
**Pull request by peterfabian**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

In addition to #28840, there are other places where we make use of compatibility check with major versions. I've discovered 2:
- update through /wp-admin/update-core.php
- force disabling auto-updates if there are any extensions that haven't declared their compatibility with WC 5.0

This change updates both places to be in line with the changes done in #28840 and double closes https://github.com/woocommerce/woocommerce/issues/28806.

### How to test the changes in this Pull Request:

1. Install one of the extensions that haven't declared compatibility with WC 5, e.g. Bookings
2. Go to /wp-admin/plugins.php
3. You should see Auto updates disabled on the right side of WC core.
4. Set WC core version to something below 5.0 (e.g 4.9.0 in woocommerce.php comment at the top and in class-woocommerce.php).
5. Go to /wp-admin/update-core.php, select WooCommerce in the list of plugins and hit update, you should see a warning as in the picture below.
6. Apply this branch.
7. Go to /wp-admin/plugins.php
3. It should be possible to enable auto-updates for WC core.
4. Go to /wp-admin/update-core.php, select WooCommerce in the list of plugins and hit update, no more warnings should stop you from updating. 

Screenshot 2021-02-23 at 13 22 08

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enhancement - Clean up of major version update compatibility warnings.

### CODE REVIEW

File: includes/admin/plugin-updates/class-wc-updates-screen-updates.php [hunk 1]
```diff
@@ -37,8 +39,13 @@ public function update_screen_modal() {
 			return;
 		}
 
+		$version_type = Constants::get_constant( 'WC_SSR_PLUGIN_UPDATE_RELEASE_VERSION_TYPE' );
```
Comment (vedanshujain) [inline includes/admin/plugin-updates/class-wc-updates-screen-updates.php:42]: This constant is currently defined only in `html-admin-page-status-report.php`, I guess this won't be defined in most flows, which means this `get_constant` call is essentially a no-op (unless you are on the SSR page)? 

Why not move this definition to wocommerce.php or skip the constant check?

File: includes/wc-core-functions.php [hunk 2]
```diff
@@ -2255,7 +2255,11 @@ function wc_prevent_dangerous_auto_updates( $should_update, $plugin ) {
 
 	$new_version      = wc_clean( $plugin->new_version );
 	$plugin_updates   = new WC_Plugin_Updates();
-	$untested_plugins = $plugin_updates->get_untested_plugins( $new_version, 'major' );
+	$version_type = Constants::get_constant( 'WC_SSR_PLUGIN_UPDATE_RELEASE_VERSION_TYPE' );
```
Comment (vedanshujain) [inline includes/wc-core-functions.php:2258]: same question about the constant here as well.

**Review by vedanshujain — COMMENTED** [review]
Looks good and testing well. I have added couple of questions because in some places code might be no-op.

I have different larger question, what happens to plugins who have declared compatibility til 3.x only. With the current code (and from the previous linked PR) seems like we won't warn against them. Is this intentional?

File: includes/admin/plugin-updates/class-wc-updates-screen-updates.php (hunk 1 above)
Comment (peterfabian) [reply includes/admin/plugin-updates/class-wc-updates-screen-updates.php:42]: You're right, thanks for noticing. I wrongly assumed this is more global. But then, on the other hand, it specifically mentions SSR, so don't know if it should actually be reused elsewhere. Does it make sense for the version check to be more granular (e.g. to only hide the warnings in SSR, but show them in the plugin screen)? I can't see how that would be useful to anyone, so I'd lean towards pulling this one out of this file to a global file. We need to preserve the name for sake of bw compatibility, though, so not great, but will add docs.

**Comment by peterfabian** [comment]
> I have different larger question, what happens to plugins who have declared compatibility til 3.x only. With the current code (and from the previous linked PR) seems like we won't warn against them. Is this intentional?

Good question! Afaik, there wasn't any truly breaking change in 4.0 (even though we were officially using SemVer back then), so I'd assume this should not be a big problem. Do you see potential for plugins breaking due to this change? 
Where I think it might actually be a problem is with plugins only declaring their compatibility with 2.x, but then we're going quite a bit back to the past.

File: includes/admin/plugin-updates/class-wc-updates-screen-updates.php (hunk 1 above)
Comment (peterfabian) [reply includes/admin/plugin-updates/class-wc-updates-screen-updates.php:42]: Please see 8566a79.

**Comment by peterfabian** [comment]
Thanks for the first round, @vedanshujain. This should be ready for another go.

**Review by vedanshujain — APPROVED** [review]
LGTM!

**Pull request merged**